src/test/java/org/simulator/sbml/SBMLTestSuiteTest.java

```diff
@@ -159,7 +165,73 @@ public void testModel() throws FileNotFoundException, IOException, XMLStreamExce
+                if (model.getExtension(CompConstants.shortLabel) != null) {
+
+                    // initialize simulator
+                    CompSimulator compSimulator = null;
+                    boolean errorInCompSimulator = false;
+                    try {
+                        compSimulator = new CompSimulator(sbmlFile);
+                    } catch (Exception e) {
+                        errorInCompSimulator = true;
+                    }
+                    Assert.assertNotNull(compSimulator);
+                    Assert.assertFalse(errorInCompSimulator);
+
+                    //solve
+                    MultiTable solution = null;
+                    boolean errorInSolve = false;
+                    try {
+                        double stepSize = (duration / steps);
+                        solution = compSimulator.solve(stepSize, duration);
+                    } catch (Exception e) {
+                        errorInSolve = true;
+                    }
+                    Assert.assertNotNull(solution);
+                    Assert.assertFalse(errorInSolve);
+
+                    //TODO: Add quality measure to check whether solution meets the correct results
+
+                } else if (model.getExtension(FBCConstants.shortLabel) != null) {
+
+                    FluxBalanceAnalysis solver = null;
+                    boolean errorInFBASimulator = false;
+                    try {
+                        solver = new FluxBalanceAnalysis(org.sbml.jsbml.SBMLReader.read(sbmlFile));
+                    } catch (Exception e) {
+                        errorInFBASimulator = true;
+                    }
+                    Assert.assertNotNull(solver);
+                    Assert.assertFalse(errorInFBASimulator);
+
+                    boolean errorInSolve = false;
+                    try {
+                        solver.solve();
+                    } catch (Exception e) {
+                        errorInSolve = true;
+                    }
+                    Assert.assertFalse(errorInSolve);
+
+                    Map<String, Double> solution = solver.getSolution();
+
+                    System.out.println(solution);
+
+                    BufferedReader reader = new BufferedReader(new FileReader(csvfile));
+                    String[] keys = reader.readLine().trim().split(",");
+                    String[] values = reader.readLine().trim().split(",");
+
+                    Map<String, Double> inputSolution = new HashMap<>();
+                    for (int i = 0; i < keys.length; i++) {
+                        inputSolution.put(keys[i], Double.valueOf(values[i]));
+                    }
+
+                    for (Map.Entry<String, Double> mapElement : inputSolution.entrySet()) {
+                        if (solution.containsKey(mapElement.getKey())) {
+                            Assert.assertEquals(mapElement.getValue(), solution.get(mapElement.getKey()), 0.0002);
```

Comment:
Whatever this magical number means (0.0002) - I would exclude it to a global var.